iset.mm

```diff
@@ -44466,6 +44466,21 @@ We use their notation ("onto" under the arrow).  (Contributed by NM,
       ( cvv wcel cop csn wfun funsng mp2an ) AEFBEFABGHICDABEEJK $.
   $}
 
+  ${
+    $d A x y z $.  $d B x y z $.
+
+    $( A function based on the singleton of an ordered pair.  Unlike ~ funsng ,
+       this holds even if ` A ` or ` B ` is a proper class.  (Contributed by
+       Jim Kingdon, 17-Apr-2022.) $)
+    funinsn $p |- Fun ( { <. A , B >. } i^i ( _V X. _V ) ) $=
```

Comment:
It might be as easy to prove
```
funinsn $p |- Fun ( { <. A , B >. } i^i ( V X. W ) ) $=
```
?

Reply:
Are you thinking of the `𝐴 ∈ 𝑉 ∧ 𝐵 ∈ 𝑊` case? Because then there is no need to intersect with anything: http://us.metamath.org/ileuni/funsng.html

Reply:
No, actually I was thinking of what I wrote ;)
But this is probably not intuitionistic, since the proof I was thinking of is a case analysis: classically, the intersection is either empty (so is a function), or is the singleton `{ <. A , B >. }`, but this happens only if A \in V and B \in W, so we can use funsng.

Reply:
> actually I was thinking of what I wrote ;)

Where `V` and `W` are arbitrary classes (not otherwise mentioned in the theorem)? I suppose that follows from funinsn from this pull request and http://us.metamath.org/ileuni/funin.html .

I agree the obvious proofs here are via case elimination. That's why I thought it was cool I found a proof of funinsn without it.

Reply:
(proved in #2590)